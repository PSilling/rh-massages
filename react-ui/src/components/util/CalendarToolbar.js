// react imports
import React from "react";
import PropTypes from "prop-types";

// module imports
import moment from "moment";

// util imports
import _t from "../../util/Translations";

/**
 * Navigation toolbar for event calendar.
 */
const CalendarToolbar = function CalendarToolbar(props) {
  return (
    <div>
      <div className="row">
        <div className="col-md-1">
          <button
            type="button"
            className="btn btn-default"
            title={_t.translate("Previous")}
            onClick={props.leftAction}
            disabled={props.leftDisabled}
          >
            <span className="glyphicon glyphicon-chevron-left" />
          </button>
        </div>

        <div className="col-md-10 text-center">
          <div className="btn-group">
            <button
              type="button"
              className={`btn btn-default${!props.monthActive ? " active" : ""}`}
              onClick={() => props.onViewChange("work_week")}
            >
              {_t.translate("Week")}
            </button>

            <button
              type="button"
              className={`btn btn-default${props.monthActive ? " active" : ""}`}
              onClick={() => props.onViewChange("month")}
            >
              {_t.translate("Month")}
            </button>
          </div>
        </div>

        <div className="col-md-1 text-right">
          <button
            type="button"
            className="btn btn-default"
            title={_t.translate("Next")}
            onClick={props.rightAction}
            disabled={props.rightDisabled}
          >
            <span className="glyphicon glyphicon-chevron-right" />
          </button>
        </div>
      </div>
      <div className="text-center" style={{ marginBottom: "5px", marginTop: "5px" }}>
        <strong>{props.month}</strong>
      </div>
    </div>
  );
};

CalendarToolbar.propTypes = {
  /** function called on left chevron button click */
  leftAction: PropTypes.func.isRequired,
  /** whether the left chevron button should be disabled */
  leftDisabled: PropTypes.bool.isRequired,
  /** whether month is the currently selected view */
  monthActive: PropTypes.bool.isRequired,
  /** function called on view change */
  onViewChange: PropTypes.func.isRequired,
  /** function called on right chevron button click */
  rightAction: PropTypes.func.isRequired,
  /** whether the right chevron button should be disabled */
  rightDisabled: PropTypes.bool.isRequired,
  /** current selected montmoment(this.state.date).format("MMMM YYYY")h */
  month: PropTypes.string
};

CalendarToolbar.defaultProps = {
  month: moment().format("MMMM YYYY")
};

export default CalendarToolbar;
