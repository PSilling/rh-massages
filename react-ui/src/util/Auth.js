// Auth handler using Keycloak
import Keycloak from 'keycloak-js';

// util imports
import Util from './Util';

/**
 * Keycloak authorization holder. Functions called on Auth work with the Keycloak
 * conncetion variable.
 */
var Auth = function() {};

Auth.keycloak = Keycloak();

/**
 * Register new and update old Clients after Keycloak authorization success.
 */
Auth.keycloak.onAuthSuccess = () => {
  Util.get(Util.CLIENTS_URL + 'my/subscribed', (json) => {
    Auth.subscribed = json;
    Util.put(Util.CLIENTS_URL, Auth.getClient(), () => {
    }, false);
  });
}

Auth.subscribed = true;

Auth.isAuthenticated = () => {
  return Auth.keycloak.authenticated;
}

Auth.isAdmin = () => {
  return Auth.keycloak.hasRealmRole("admin");
}

Auth.getToken = () => {
  return Auth.keycloak.token;
}

Auth.getSub = () => {
  return Auth.keycloak.subject;
}

/**
 * Generates a new Client based on server Client representation.
 */
Auth.getClient = () => {
  return {
    sub: Auth.keycloak.subject, email: Auth.keycloak.idTokenParsed.email,
    name: Auth.keycloak.idTokenParsed.given_name, surname: Auth.keycloak.idTokenParsed.family_name,
    subscribed: Auth.subscribed
  };
}

export default Auth
