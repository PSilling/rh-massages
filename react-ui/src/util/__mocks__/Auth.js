const isAdmin = jest.fn().mockReturnValue(true);
const isAuthenticated = jest.fn().mockReturnValue(true);
const subscribed = true;
const getToken = jest.fn().mockReturnValue("secret");
const getSub = jest.fn().mockReturnValue("test");
const getClient = jest.fn().mockReturnValue({
  sub: "test", email: "test@email.com",
  name: "Test", surname: "User",
  subscribed: true
});

module.exports = {
  isAdmin,
  isAuthenticated,
  subscribed,
  getToken,
  getSub,
  getClient
}
