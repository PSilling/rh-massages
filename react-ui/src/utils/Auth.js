// Auth "service"
const Auth = {

  isAuthenticated() {
    return true;
  },

  isAdmin() {
    return true;
  }
}

export default Auth
