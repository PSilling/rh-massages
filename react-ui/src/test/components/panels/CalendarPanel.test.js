// react imports
import React from "react";

// test imports
import { shallow } from "enzyme";
import BigCalendar from "react-big-calendar";
import Auth from "../../../util/Auth";
import CalendarPanel from "../../../components/panels/CalendarPanel";
import CalendarToolbar from "../../../components/util/CalendarToolbar";
import MassageEventModal from "../../../components/modals/MassageEventModal";

// test mocks
jest.mock("../../../util/Auth");

afterEach(() => {
  jest.resetAllMocks();
});

test("renders content with correct props and funcionality", () => {
  const testAssignFunction = jest.fn();
  const testCancelFunction = jest.fn();
  const testAddFunction = jest.fn();
  const testEditFunction = jest.fn();
  const testDeleteFunction = jest.fn();
  const testDateChangeFunction = jest.fn();
  const testSelectFunction = jest.fn();
  const testEvents = [
    {
      bgColor: "white",
      massage: {
        id: 1,
        date: new Date(0),
        ending: new Date(1000),
        masseuse: "test",
        client: null,
        facility: { id: 1, name: "test" }
      }
    },
    {
      bgColor: "black",
      massage: {
        id: 2,
        date: new Date(1000),
        ending: new Date(2000),
        client: { sub: "test" },
        facility: { id: 1, name: "test" }
      }
    },
    {
      bgColor: "yellow",
      massage: {
        id: 3,
        date: new Date(2000),
        ending: new Date(3000),
        client: { sub: "test2" },
        facility: { id: 1, name: "test" }
      }
    }
  ];
  const wrapper = shallow(
    <CalendarPanel
      events={testEvents}
      selected={[]}
      selectEvents
      massageMinutes={0}
      onAssign={testAssignFunction}
      onCancel={testCancelFunction}
      onAdd={testAddFunction}
      onEdit={testEditFunction}
      onDelete={testDeleteFunction}
      onDateChange={testDateChangeFunction}
      onSelect={testSelectFunction}
    />
  );

  const toolbar = wrapper.find(CalendarToolbar);
  const calendar = wrapper.find(BigCalendar);

  expect(toolbar.props().leftDisabled).toBe(true);
  expect(toolbar.props().rightDisabled).toBe(false);
  wrapper.instance().changeDate(false);
  expect(testDateChangeFunction).toHaveBeenCalledTimes(1);
  wrapper.instance().changeDate(true);
  expect(testDateChangeFunction).toHaveBeenCalledTimes(2);
  wrapper.instance().changeView("month");
  expect(testDateChangeFunction).toHaveBeenCalledTimes(3);
  expect(testDateChangeFunction).toHaveBeenLastCalledWith(wrapper.instance().state.date, "month");

  expect(calendar.props().messages).toBe(wrapper.instance().localization);
  expect(calendar.props().events).toBe(testEvents);
  expect(calendar.props().views).toEqual(["work_week", "month"]);
  calendar.props().onView(null);
  calendar.props().onNavigate(null, null);
  expect(testDateChangeFunction).toHaveBeenCalledTimes(5);
  expect(calendar.props().titleAccessor(testEvents[0])).toBe(testEvents[0].massage.masseuse);
  expect(calendar.props().startAccessor(testEvents[0])).toEqual(testEvents[0].massage.date);
  expect(calendar.props().endAccessor(testEvents[0])).toEqual(testEvents[0].massage.ending);
  expect(calendar.props().onSelectSlot).toBe(testAddFunction);

  wrapper.instance().configureModalActions(testEvents[0]);
  expect(wrapper.instance().state.action).toBe("assign");
  wrapper.instance().configureModalActions(testEvents[1]);
  expect(wrapper.instance().state.action).toBe("cancel");
  wrapper.instance().configureModalActions(testEvents[2]);
  expect(wrapper.instance().state.action).toBe("cancel");
  Auth.isAdmin = jest.fn(() => false);
  wrapper.instance().configureModalActions(testEvents[2]);
  expect(wrapper.instance().state.action).toBe("none");

  const modal = wrapper.find(MassageEventModal);

  expect(modal.props().event).toBe(testEvents[2]);
  expect(modal.props().label).toBe("none");
  expect(modal.props().disabled).toBe(false);
  expect(modal.props().allowEditation).toBe(true);
  modal.props().onEdit();
  modal.props().onDelete();
  expect(testEditFunction).toHaveBeenLastCalledWith(testEvents[2].massage);
  expect(testDeleteFunction).toHaveBeenLastCalledWith(testEvents[2].massage.id);
  expect(modal.props().onClose).toBe(wrapper.instance().handleToggle);

  expect(wrapper).toMatchSnapshot();
});
