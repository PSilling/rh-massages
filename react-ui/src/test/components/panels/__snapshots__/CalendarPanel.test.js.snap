// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props and funcionality 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CalendarPanel
    allowEditation={true}
    events={
        Array [
            Object {
              "bgColor": "white",
              "massage": Object {
                "client": null,
                "date": 1970-01-01T00:00:00.000Z,
                "ending": 1970-01-01T00:00:01.000Z,
                "facility": Object {
                  "id": 1,
                  "name": "test",
                },
                "id": 1,
                "masseuse": "test",
              },
            },
            Object {
              "bgColor": "black",
              "massage": Object {
                "client": Object {
                  "sub": "test",
                },
                "date": 1970-01-01T00:00:01.000Z,
                "ending": 1970-01-01T00:00:02.000Z,
                "facility": Object {
                  "id": 1,
                  "name": "test",
                },
                "id": 2,
              },
            },
            Object {
              "bgColor": "yellow",
              "massage": Object {
                "client": Object {
                  "sub": "test2",
                },
                "date": 1970-01-01T00:00:02.000Z,
                "ending": 1970-01-01T00:00:03.000Z,
                "facility": Object {
                  "id": 1,
                  "name": "test",
                },
                "id": 3,
              },
            },
          ]
    }
    massageMinutes={0}
    onAdd={[Function]}
    onAssign={[Function]}
    onCancel={[Function]}
    onDateChange={[Function]}
    onDelete={[Function]}
    onEdit={[Function]}
    onSelect={[Function]}
    selectEvents={true}
    selected={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CalendarToolbar
          leftAction={[Function]}
          leftDisabled={false}
          month="Invalid date"
          monthActive={true}
          onViewChange={[Function]}
          rightAction={[Function]}
          rightDisabled={false}
/>,
        <Uncontrolled(Calendar)
          date={null}
          endAccessor={[Function]}
          eventPropGetter={[Function]}
          events={
                    Array [
                              Object {
                                "bgColor": "white",
                                "massage": Object {
                                  "client": null,
                                  "date": 1970-01-01T00:00:00.000Z,
                                  "ending": 1970-01-01T00:00:01.000Z,
                                  "facility": Object {
                                    "id": 1,
                                    "name": "test",
                                  },
                                  "id": 1,
                                  "masseuse": "test",
                                },
                              },
                              Object {
                                "bgColor": "black",
                                "massage": Object {
                                  "client": Object {
                                    "sub": "test",
                                  },
                                  "date": 1970-01-01T00:00:01.000Z,
                                  "ending": 1970-01-01T00:00:02.000Z,
                                  "facility": Object {
                                    "id": 1,
                                    "name": "test",
                                  },
                                  "id": 2,
                                },
                              },
                              Object {
                                "bgColor": "yellow",
                                "massage": Object {
                                  "client": Object {
                                    "sub": "test2",
                                  },
                                  "date": 1970-01-01T00:00:02.000Z,
                                  "ending": 1970-01-01T00:00:03.000Z,
                                  "facility": Object {
                                    "id": 1,
                                    "name": "test",
                                  },
                                  "id": 3,
                                },
                              },
                            ]
          }
          max={2018-01-01T17:00:00.000Z}
          messages={
                    Object {
                              "allDay": "Celý den",
                              "date": "Datum",
                              "event": "Událost",
                              "month": "Měsíc",
                              "next": ">",
                              "previous": "<",
                              "showMore": [Function],
                              "time": "Čas",
                              "today": "Dnes",
                              "work_week": "Týden",
                            }
          }
          min={2018-01-01T07:30:00.000Z}
          onNavigate={[Function]}
          onSelectEvent={[Function]}
          onSelectSlot={[Function]}
          onView={[Function]}
          popup={true}
          selectable={false}
          startAccessor={[Function]}
          style={
                    Object {
                              "height": "85vh",
                            }
          }
          timeslots={1}
          titleAccessor={[Function]}
          toolbar={false}
          view="month"
          views={
                    Array [
                              "work_week",
                              "month",
                            ]
          }
/>,
        <div
          className="row text-center"
          style={
                    Object {
                              "marginTop": "16px",
                            }
          }
>
          <div
                    className="col-md-12"
          >
                    <strong>
                              Legenda:
                    </strong>
                    <span
                              style={
                                        Object {
                                                  "backgroundColor": "#2fad2f",
                                                  "borderRadius": "4px",
                                                  "marginLeft": "8px",
                                                  "padding": "8px",
                                                }
                              }
                    >
                              Volná masáž
                    </span>
                    <span
                              style={
                                        Object {
                                                  "backgroundColor": "#ee9d2a",
                                                  "borderRadius": "4px",
                                                  "marginLeft": "8px",
                                                  "padding": "8px",
                                                }
                              }
                    >
                              Moje masáž
                    </span>
                    <span
                              style={
                                        Object {
                                                  "backgroundColor": "#d10a14",
                                                  "borderRadius": "4px",
                                                  "marginLeft": "8px",
                                                  "padding": "8px",
                                                }
                              }
                    >
                              Přiřazená masáž
                    </span>
          </div>
</div>,
        "",
      ],
      "style": Object {
        "marginBottom": "5vh",
        "marginTop": "10px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "leftAction": [Function],
          "leftDisabled": false,
          "month": "Invalid date",
          "monthActive": true,
          "onViewChange": [Function],
          "rightAction": [Function],
          "rightDisabled": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "date": null,
          "endAccessor": [Function],
          "eventPropGetter": [Function],
          "events": Array [
            Object {
              "bgColor": "white",
              "massage": Object {
                "client": null,
                "date": 1970-01-01T00:00:00.000Z,
                "ending": 1970-01-01T00:00:01.000Z,
                "facility": Object {
                  "id": 1,
                  "name": "test",
                },
                "id": 1,
                "masseuse": "test",
              },
            },
            Object {
              "bgColor": "black",
              "massage": Object {
                "client": Object {
                  "sub": "test",
                },
                "date": 1970-01-01T00:00:01.000Z,
                "ending": 1970-01-01T00:00:02.000Z,
                "facility": Object {
                  "id": 1,
                  "name": "test",
                },
                "id": 2,
              },
            },
            Object {
              "bgColor": "yellow",
              "massage": Object {
                "client": Object {
                  "sub": "test2",
                },
                "date": 1970-01-01T00:00:02.000Z,
                "ending": 1970-01-01T00:00:03.000Z,
                "facility": Object {
                  "id": 1,
                  "name": "test",
                },
                "id": 3,
              },
            },
          ],
          "max": 2018-01-01T17:00:00.000Z,
          "messages": Object {
            "allDay": "Celý den",
            "date": "Datum",
            "event": "Událost",
            "month": "Měsíc",
            "next": ">",
            "previous": "<",
            "showMore": [Function],
            "time": "Čas",
            "today": "Dnes",
            "work_week": "Týden",
          },
          "min": 2018-01-01T07:30:00.000Z,
          "onNavigate": [Function],
          "onSelectEvent": [Function],
          "onSelectSlot": [Function],
          "onView": [Function],
          "popup": true,
          "selectable": false,
          "startAccessor": [Function],
          "style": Object {
            "height": "85vh",
          },
          "timeslots": 1,
          "titleAccessor": [Function],
          "toolbar": false,
          "view": "month",
          "views": Array [
            "work_week",
            "month",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-md-12"
>
            <strong>
                        Legenda:
            </strong>
            <span
                        style={
                                    Object {
                                                "backgroundColor": "#2fad2f",
                                                "borderRadius": "4px",
                                                "marginLeft": "8px",
                                                "padding": "8px",
                                              }
                        }
            >
                        Volná masáž
            </span>
            <span
                        style={
                                    Object {
                                                "backgroundColor": "#ee9d2a",
                                                "borderRadius": "4px",
                                                "marginLeft": "8px",
                                                "padding": "8px",
                                              }
                        }
            >
                        Moje masáž
            </span>
            <span
                        style={
                                    Object {
                                                "backgroundColor": "#d10a14",
                                                "borderRadius": "4px",
                                                "marginLeft": "8px",
                                                "padding": "8px",
                                              }
                        }
            >
                        Přiřazená masáž
            </span>
</div>,
          "className": "row text-center",
          "style": Object {
            "marginTop": "16px",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                Legenda:
</strong>,
              <span
                style={
                                Object {
                                                "backgroundColor": "#2fad2f",
                                                "borderRadius": "4px",
                                                "marginLeft": "8px",
                                                "padding": "8px",
                                              }
                }
>
                Volná masáž
</span>,
              <span
                style={
                                Object {
                                                "backgroundColor": "#ee9d2a",
                                                "borderRadius": "4px",
                                                "marginLeft": "8px",
                                                "padding": "8px",
                                              }
                }
>
                Moje masáž
</span>,
              <span
                style={
                                Object {
                                                "backgroundColor": "#d10a14",
                                                "borderRadius": "4px",
                                                "marginLeft": "8px",
                                                "padding": "8px",
                                              }
                }
>
                Přiřazená masáž
</span>,
            ],
            "className": "col-md-12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Legenda:",
              },
              "ref": null,
              "rendered": "Legenda:",
              "type": "strong",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Volná masáž",
                "style": Object {
                  "backgroundColor": "#2fad2f",
                  "borderRadius": "4px",
                  "marginLeft": "8px",
                  "padding": "8px",
                },
              },
              "ref": null,
              "rendered": "Volná masáž",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Moje masáž",
                "style": Object {
                  "backgroundColor": "#ee9d2a",
                  "borderRadius": "4px",
                  "marginLeft": "8px",
                  "padding": "8px",
                },
              },
              "ref": null,
              "rendered": "Moje masáž",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Přiřazená masáž",
                "style": Object {
                  "backgroundColor": "#d10a14",
                  "borderRadius": "4px",
                  "marginLeft": "8px",
                  "padding": "8px",
                },
              },
              "ref": null,
              "rendered": "Přiřazená masáž",
              "type": "span",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CalendarToolbar
            leftAction={[Function]}
            leftDisabled={false}
            month="Invalid date"
            monthActive={true}
            onViewChange={[Function]}
            rightAction={[Function]}
            rightDisabled={false}
/>,
          <Uncontrolled(Calendar)
            date={null}
            endAccessor={[Function]}
            eventPropGetter={[Function]}
            events={
                        Array [
                                    Object {
                                      "bgColor": "white",
                                      "massage": Object {
                                        "client": null,
                                        "date": 1970-01-01T00:00:00.000Z,
                                        "ending": 1970-01-01T00:00:01.000Z,
                                        "facility": Object {
                                          "id": 1,
                                          "name": "test",
                                        },
                                        "id": 1,
                                        "masseuse": "test",
                                      },
                                    },
                                    Object {
                                      "bgColor": "black",
                                      "massage": Object {
                                        "client": Object {
                                          "sub": "test",
                                        },
                                        "date": 1970-01-01T00:00:01.000Z,
                                        "ending": 1970-01-01T00:00:02.000Z,
                                        "facility": Object {
                                          "id": 1,
                                          "name": "test",
                                        },
                                        "id": 2,
                                      },
                                    },
                                    Object {
                                      "bgColor": "yellow",
                                      "massage": Object {
                                        "client": Object {
                                          "sub": "test2",
                                        },
                                        "date": 1970-01-01T00:00:02.000Z,
                                        "ending": 1970-01-01T00:00:03.000Z,
                                        "facility": Object {
                                          "id": 1,
                                          "name": "test",
                                        },
                                        "id": 3,
                                      },
                                    },
                                  ]
            }
            max={2018-01-01T17:00:00.000Z}
            messages={
                        Object {
                                    "allDay": "Celý den",
                                    "date": "Datum",
                                    "event": "Událost",
                                    "month": "Měsíc",
                                    "next": ">",
                                    "previous": "<",
                                    "showMore": [Function],
                                    "time": "Čas",
                                    "today": "Dnes",
                                    "work_week": "Týden",
                                  }
            }
            min={2018-01-01T07:30:00.000Z}
            onNavigate={[Function]}
            onSelectEvent={[Function]}
            onSelectSlot={[Function]}
            onView={[Function]}
            popup={true}
            selectable={false}
            startAccessor={[Function]}
            style={
                        Object {
                                    "height": "85vh",
                                  }
            }
            timeslots={1}
            titleAccessor={[Function]}
            toolbar={false}
            view="month"
            views={
                        Array [
                                    "work_week",
                                    "month",
                                  ]
            }
/>,
          <div
            className="row text-center"
            style={
                        Object {
                                    "marginTop": "16px",
                                  }
            }
>
            <div
                        className="col-md-12"
            >
                        <strong>
                                    Legenda:
                        </strong>
                        <span
                                    style={
                                                Object {
                                                            "backgroundColor": "#2fad2f",
                                                            "borderRadius": "4px",
                                                            "marginLeft": "8px",
                                                            "padding": "8px",
                                                          }
                                    }
                        >
                                    Volná masáž
                        </span>
                        <span
                                    style={
                                                Object {
                                                            "backgroundColor": "#ee9d2a",
                                                            "borderRadius": "4px",
                                                            "marginLeft": "8px",
                                                            "padding": "8px",
                                                          }
                                    }
                        >
                                    Moje masáž
                        </span>
                        <span
                                    style={
                                                Object {
                                                            "backgroundColor": "#d10a14",
                                                            "borderRadius": "4px",
                                                            "marginLeft": "8px",
                                                            "padding": "8px",
                                                          }
                                    }
                        >
                                    Přiřazená masáž
                        </span>
            </div>
</div>,
          "",
        ],
        "style": Object {
          "marginBottom": "5vh",
          "marginTop": "10px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "leftAction": [Function],
            "leftDisabled": false,
            "month": "Invalid date",
            "monthActive": true,
            "onViewChange": [Function],
            "rightAction": [Function],
            "rightDisabled": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "date": null,
            "endAccessor": [Function],
            "eventPropGetter": [Function],
            "events": Array [
              Object {
                "bgColor": "white",
                "massage": Object {
                  "client": null,
                  "date": 1970-01-01T00:00:00.000Z,
                  "ending": 1970-01-01T00:00:01.000Z,
                  "facility": Object {
                    "id": 1,
                    "name": "test",
                  },
                  "id": 1,
                  "masseuse": "test",
                },
              },
              Object {
                "bgColor": "black",
                "massage": Object {
                  "client": Object {
                    "sub": "test",
                  },
                  "date": 1970-01-01T00:00:01.000Z,
                  "ending": 1970-01-01T00:00:02.000Z,
                  "facility": Object {
                    "id": 1,
                    "name": "test",
                  },
                  "id": 2,
                },
              },
              Object {
                "bgColor": "yellow",
                "massage": Object {
                  "client": Object {
                    "sub": "test2",
                  },
                  "date": 1970-01-01T00:00:02.000Z,
                  "ending": 1970-01-01T00:00:03.000Z,
                  "facility": Object {
                    "id": 1,
                    "name": "test",
                  },
                  "id": 3,
                },
              },
            ],
            "max": 2018-01-01T17:00:00.000Z,
            "messages": Object {
              "allDay": "Celý den",
              "date": "Datum",
              "event": "Událost",
              "month": "Měsíc",
              "next": ">",
              "previous": "<",
              "showMore": [Function],
              "time": "Čas",
              "today": "Dnes",
              "work_week": "Týden",
            },
            "min": 2018-01-01T07:30:00.000Z,
            "onNavigate": [Function],
            "onSelectEvent": [Function],
            "onSelectSlot": [Function],
            "onView": [Function],
            "popup": true,
            "selectable": false,
            "startAccessor": [Function],
            "style": Object {
              "height": "85vh",
            },
            "timeslots": 1,
            "titleAccessor": [Function],
            "toolbar": false,
            "view": "month",
            "views": Array [
              "work_week",
              "month",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-md-12"
>
              <strong>
                            Legenda:
              </strong>
              <span
                            style={
                                          Object {
                                                        "backgroundColor": "#2fad2f",
                                                        "borderRadius": "4px",
                                                        "marginLeft": "8px",
                                                        "padding": "8px",
                                                      }
                            }
              >
                            Volná masáž
              </span>
              <span
                            style={
                                          Object {
                                                        "backgroundColor": "#ee9d2a",
                                                        "borderRadius": "4px",
                                                        "marginLeft": "8px",
                                                        "padding": "8px",
                                                      }
                            }
              >
                            Moje masáž
              </span>
              <span
                            style={
                                          Object {
                                                        "backgroundColor": "#d10a14",
                                                        "borderRadius": "4px",
                                                        "marginLeft": "8px",
                                                        "padding": "8px",
                                                      }
                            }
              >
                            Přiřazená masáž
              </span>
</div>,
            "className": "row text-center",
            "style": Object {
              "marginTop": "16px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <strong>
                  Legenda:
</strong>,
                <span
                  style={
                                    Object {
                                                      "backgroundColor": "#2fad2f",
                                                      "borderRadius": "4px",
                                                      "marginLeft": "8px",
                                                      "padding": "8px",
                                                    }
                  }
>
                  Volná masáž
</span>,
                <span
                  style={
                                    Object {
                                                      "backgroundColor": "#ee9d2a",
                                                      "borderRadius": "4px",
                                                      "marginLeft": "8px",
                                                      "padding": "8px",
                                                    }
                  }
>
                  Moje masáž
</span>,
                <span
                  style={
                                    Object {
                                                      "backgroundColor": "#d10a14",
                                                      "borderRadius": "4px",
                                                      "marginLeft": "8px",
                                                      "padding": "8px",
                                                    }
                  }
>
                  Přiřazená masáž
</span>,
              ],
              "className": "col-md-12",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Legenda:",
                },
                "ref": null,
                "rendered": "Legenda:",
                "type": "strong",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Volná masáž",
                  "style": Object {
                    "backgroundColor": "#2fad2f",
                    "borderRadius": "4px",
                    "marginLeft": "8px",
                    "padding": "8px",
                  },
                },
                "ref": null,
                "rendered": "Volná masáž",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Moje masáž",
                  "style": Object {
                    "backgroundColor": "#ee9d2a",
                    "borderRadius": "4px",
                    "marginLeft": "8px",
                    "padding": "8px",
                  },
                },
                "ref": null,
                "rendered": "Moje masáž",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Přiřazená masáž",
                  "style": Object {
                    "backgroundColor": "#d10a14",
                    "borderRadius": "4px",
                    "marginLeft": "8px",
                    "padding": "8px",
                  },
                },
                "ref": null,
                "rendered": "Přiřazená masáž",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
